File: backend/src/main/java/ma/zs/alc/service/facade/admin/vocab/CollaboratorAdminService.java
package ma.zs.alc.service.facade.admin.vocab;

import ma.zs.alc.bean.core.common.Collaborator;
import ma.zs.alc.dao.criteria.core.vocab.CollaboratorCriteria;
import ma.zs.alc.dao.facade.core.inscription.UserPageable;
import ma.zs.alc.zynerator.service.IService;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;

import java.util.List;


public interface CollaboratorAdminService extends IService<Collaborator, CollaboratorCriteria> {
    Collaborator findByUsername(String username);

    boolean changePassword(String username, String newPassword);

    List<Collaborator> findByTypeCollaboratorId(Long id);

    int deleteByTypeCollaboratorId(Long id);

    long countByTypeCollaboratorCode(String code);

    Page<UserPageable> findAllByCriteria(Pageable pageable);
}

File: frontend/src/app/shared/service/admin/vocab/CollaboratorAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {CollaboratorDto} from '../../../model/vocab/Collaborator.model';
import {CollaboratorCriteria} from '../../../criteria/vocab/CollaboratorCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";
import {Pageable} from "../../../utils/Pageable";
import {Observable} from "rxjs";
import {ManageUserDto} from "../../../../core/criteria/manage-user-dto";
import {PageRequest} from "../../../../zynerator/criteria/BaseCriteria.model";
import {ProfDto} from "../../../model/prof/Prof.model";
import {EtudiantDto} from "../../../model/inscription/Etudiant.model";


@Injectable({
    providedIn: 'root'
})
export class CollaboratorAdminService extends AbstractService<CollaboratorDto, CollaboratorCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/collaborator/';
    }

    public constrcutDto(): CollaboratorDto {
        return new CollaboratorDto();
    }

    public constrcutCriteria(): CollaboratorCriteria {
        return new CollaboratorCriteria();
    }

    findAllByCriteria(pageable: Pageable): Observable<PageRequest<ManageUserDto>> {
        return this.http.get<PageRequest<ManageUserDto>>(this.API + `pageable`,
            {
                params: {
                    'page': pageable.page,
                    'size': pageable.size
                }
            });
    }

    findAllOptimized(): Observable<Array<CollaboratorDto>> {
        return this.http.get<Array<CollaboratorDto>>(this.API + `optimized`);
    }

    update(user: CollaboratorDto): Observable<CollaboratorDto> {
        return this.http.put<CollaboratorDto>(this.API, user);
    }
    create(user: CollaboratorDto): Observable<CollaboratorDto> {
        return this.http.post<CollaboratorDto>(this.API + 'create', user);
    }
    deleteById(id: number): Observable<number> {
        return this.http.delete<number>(`${this.API}id/${id}`);
    }
}
